### ISSUE TITLE
Train action recognition

### ISSUE THREAD

**Issue by reddevil1310** [dmlc/gluon-cv#1678]
I follow this tutorial : https://cv.gluon.ai/build/examples_action_recognition/dive_deep_i3d_kinetics400.html
I use some class in kinetics700. 
I saw no increase in accuracy through loops as below. Can you tell me if I'm doing something wrong here? 
Thank you so much!

[Epoch 7] train=0.118812 loss=2.853746 time: 334.248697
[Epoch 8] train=0.120792 loss=2.800172 time: 346.450900
[Epoch 9] train=0.104950 loss=2.824681 time: 320.679160
[Epoch 10] train=0.132673 loss=2.823033 time: 327.287522
[Epoch 11] train=0.100990 loss=2.805340 time: 361.680605
[Epoch 12] train=0.116832 loss=2.829458 time: 353.433977
[Epoch 13] train=0.126733 loss=2.806885 time: 348.658716
[Epoch 14] train=0.118812 loss=2.795670 time: 318.316561
[Epoch 15] train=0.112871 loss=2.801041 time: 328.427944
[Epoch 16] train=0.124752 loss=2.794525 time: 321.472601
[Epoch 17] train=0.130693 loss=2.774280 time: 362.312669
[Epoch 18] train=0.154455 loss=2.739297 time: 357.701154
[Epoch 19] train=0.120792 loss=2.803970 time: 478.908508
[Epoch 20] train=0.140594 loss=2.772666 time: 1705.808573
[Epoch 21] train=0.158416 loss=2.776253 time: 1714.771228
[Epoch 22] train=0.114851 loss=2.792731 time: 1657.656340
[Epoch 23] train=0.126733 loss=2.772835 time: 1771.440408
[Epoch 24] train=0.148515 loss=2.736776 time: 1417.960037
[Epoch 25] train=0.122772 loss=2.756343 time: 385.093622
[Epoch 26] train=0.142574 loss=2.794799 time: 336.805288

**Comment by tcfkaj** [comment]
This is not really the appropriate place for a question like this. Issues are meant for topics directly involving the package. Your problem could be anything from just needing to tweak your parameters to an issue with your dataset. A more appropriate place to ask this would be [discuss.mxnet.apache.org/](https://discuss.mxnet.apache.org/) or [discuss.pytorch.org](https://discuss.pytorch.org/) , depending on what backend you're using.

Additionally, if you want a meaningful answer you should provide your code and different things you have tried to troubleshoot so far.

**Comment by bryanyzhu** [comment]
@tcfkaj Thank you for your clarification. 

@reddevil1310 I saw the training time of each epoch is really fast. How many classes and training samples do you have? If you are training on a small dataset, you may need to freeze the backbone and only update the classification head, or at least freeze all the BN layers except for the first one. Otherwise, the training will not converge. 

Also, as @tcfkaj pointed out, in order to help debug this issue, you might need to provide more information. Many little details will effect the training, e.g., learning rate, init_std in the last fc layer, weight decay, etc.

**Comment by reddevil1310** [comment]
@bryanyzhu I use 20 classes of Kinetics700. I have downloaded all the data of these 20 classes. the network parameters i configured as in this tutorial: https://cv.gluon.ai/build/examples_action_recognition/dive_deep_i3d_kinetics400.html
Sorry but I don't understand how to freeze the back bone.
@tcfkaj Thank you.

**Comment by reddevil1310** [comment]
@bryanyzhu  In this tutorial : https://cv.gluon.ai/build/examples_action_recognition/dive_deep_i3d_kinetics400.html, I wonder why only use 100 samples in train data, break if i ==100
    # Loop through each batch of training data
    for i, batch in enumerate(train_data):
        # Extract data and label
        data = split_and_load(batch[0], ctx_list=ctx, batch_axis=0)
        label = split_and_load(batch[1], ctx_list=ctx, batch_axis=0)

        # AutoGrad
        with ag.record():
            output = []
            for _, X in enumerate(data):
                X = X.reshape((-1,) + X.shape[2:])
                pred = net(X)
                output.append(pred)
            loss = [loss_fn(yhat, y) for yhat, y in zip(output, label)]

        # Backpropagation
        for l in loss:
            l.backward()

        # Optimize
        trainer.step(batch_size)

        # Update metrics
        train_loss += sum([l.mean().asscalar() for l in loss])
        train_metric.update(label, output)

        if i == 100:
            break

**Comment by bryanyzhu** [comment]
@reddevil1310 `i` is the training iteration number, which means I only want to train the network for 100 iterations. It is not just using 100 training samples (For example, if the training batch size is 16, 100 iterations equal to 1600 training samples.) The reason I choose 100 is simply for demo purpose. I don't want to finish the training on our CI server.

**Comment by reddevil1310** [comment]
Thank you very much for your help. After use pretrained weights of Kinetics400, Accuracy has increased over 50%.
